### PR TITLE
IMPORTANT: Added Apache License

### PR DESCRIPTION
**Pull request by ricdezen**
I added the Apache licence because we still were missing a license at all. If anyone has a more restrictive wish about how we should license our work, we should discuss it asap.

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+          as modifying the License.
+
+      You may add Your own copyright statement to Your modifications and
+      may provide additional or different license terms and conditions
+      for use, reproduction, or distribution of Your modifications, or
+      for any such Derivative Works as a whole, provided Your use,
+      reproduction, and distribution of the Work otherwise complies with
+      the conditions stated in this License.
+
+   5. Submission of Contributions. Unless You explicitly state otherwise,
```
Comment (ricdezen) [inline LICENSE.md:130]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

**Comment by ricdezen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 27
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568&bid=16390186)

File: LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
+
+   TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
+
+   1. Definitions.
```
Comment (ricdezen) [inline LICENSE.md:7]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      Notwithstanding the above, nothing herein shall supersede or modify
+      the terms of any separate license agreement you may have executed
+      with Licensor regarding such Contributions.
+
+   6. Trademarks. This License does not grant permission to use the trade
+      names, trademarks, service marks, or product names of the Licensor,
+      except as required for reasonable and customary use in describing the
+      origin of the Work and reproducing the content of the NOTICE file.
+
+   7. Disclaimer of Warranty. Unless required by applicable law or
```
Comment (ricdezen) [inline LICENSE.md:143]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (ricdezen) [inline LICENSE.md:189]: Codacy Issue found: [[list-item-content-indent] Don’t use mixed indentation for children, add 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
```
Comment (ricdezen) [inline LICENSE.md:178]: Codacy Issue found: [[list-item-content-indent] Don’t use mixed indentation for children, add 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
+
+   Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (ricdezen) [inline LICENSE.md:191]: Codacy Issue found: [[list-item-content-indent] Don’t use mixed indentation for children, add 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+      agreed to in writing, Licensor provides the Work (and each
+      Contributor provides its Contributions) on an "AS IS" BASIS,
+      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+      implied, including, without limitation, any warranties or conditions
+      of TITLE, NON-INFRINGEMENT, MERCHANTABILITY, or FITNESS FOR A
+      PARTICULAR PURPOSE. You are solely responsible for determining the
+      appropriateness of using or redistributing the Work and assume any
+      risks associated with Your exercise of permissions under this License.
+
+   8. Limitation of Liability. In no event and under no legal theory,
```
Comment (ricdezen) [inline LICENSE.md:153]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md (hunk 3 above)
Comment (ricdezen) [inline LICENSE.md:143]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md (hunk 7 above)
Comment (ricdezen) [inline LICENSE.md:153]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+      negligent acts) or agreed to in writing, shall any Contributor be
+      liable to You for damages, including any direct, indirect, special,
+      incidental, or consequential damages of any character arising as a
+      result of this License or out of the use or inability to use the
+      Work (including but not limited to damages for loss of goodwill,
+      work stoppage, computer failure or malfunction, or any and all
+      other commercial damages or losses), even if such Contributor
+      has been advised of the possibility of such damages.
+
+   9. Accepting Warranty or Additional Liability. While redistributing
```
Comment (ricdezen) [inline LICENSE.md:165]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+      Contribution(s) alone or by combination of their Contribution(s)
+      with the Work to which such Contribution(s) was submitted. If You
+      institute patent litigation against any entity (including a
+      cross-claim or counterclaim in a lawsuit) alleging that the Work
+      or a Contribution incorporated within the Work constitutes direct
+      or contributory patent infringement, then any patent licenses
+      granted to You under this License for that Work shall terminate
+      as of the date such litigation is filed.
+
+   4. Redistribution. You may reproduce and distribute copies of the
```
Comment (ricdezen) [inline LICENSE.md:89]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

Comment (ricdezen) [inline LICENSE.md:89]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+   5. Submission of Contributions. Unless You explicitly state otherwise,
+      any Contribution intentionally submitted for inclusion in the Work
+      by You to the Licensor shall be under the terms and conditions of
+      this License, without any additional terms or conditions.
+      Notwithstanding the above, nothing herein shall supersede or modify
+      the terms of any separate license agreement you may have executed
+      with Licensor regarding such Contributions.
+
+   6. Trademarks. This License does not grant permission to use the trade
```
Comment (ricdezen) [inline LICENSE.md:138]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md (hunk 4 above)
Comment (ricdezen) [inline LICENSE.md:189]: Codacy Issue found: [[no-undefined-references] Found reference to undefined definition](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+      subsequently incorporated within the Work.
+
+   2. Grant of Copyright License. Subject to the terms and conditions of
+      this License, each Contributor hereby grants to You a perpetual,
+      worldwide, non-exclusive, no-charge, royalty-free, irrevocable
+      copyright license to reproduce, prepare Derivative Works of,
+      publicly display, publicly perform, sublicense, and distribute the
+      Work and such Derivative Works in Source or Object form.
+
+   3. Grant of Patent License. Subject to the terms and conditions of
```
Comment (ricdezen) [inline LICENSE.md:73]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
+
+   APPENDIX: How to apply the Apache License to your work.
+
+      To apply the Apache License to your work, attach the following
```
Comment (ricdezen) [inline LICENSE.md:180]: Codacy Issue found: [[list-item-content-indent] Don’t use mixed indentation for children, add 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md (hunk 8 above)
Comment (ricdezen) [inline LICENSE.md:165]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+      and charge a fee for, acceptance of support, warranty, indemnity,
+      or other liability obligations and/or rights consistent with this
+      License. However, in accepting such obligations, You may act only
+      on Your own behalf and on Your sole responsibility, not on behalf
+      of any other Contributor, and only if You agree to indemnify,
+      defend, and hold each Contributor harmless for any liability
+      incurred by, or claims asserted against, such Contributor by reason
+      of your accepting any such warranty or additional liability.
+
+   END OF TERMS AND CONDITIONS
```
Comment (ricdezen) [inline LICENSE.md:176]: Codacy Issue found: [[list-item-content-indent] Don’t use mixed indentation for children, add 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: README.md [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+The content of this repository is the work of students of the "Elementi di Ingegneria del Software" course held at the University of Padova during the first semester of the 2019/2020 academic year.
+
+## License
+The work is licensed under the Apache License, a copy of which is located in the [LICENSE.md](LICENSE.md) file.
+
+Copyright 2020 Riccardo De Zen, Niccolò Turcato, Giorgia Bortoletti, Pardeep Kumar.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
```
Comment (ricdezen) [inline README.md:11]: Codacy Issue found: [[no-literal-urls] Don’t use literal URLs without angle brackets](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 15]
```diff
@@ -0,0 +1,201 @@
...
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
```
Comment (ricdezen) [inline LICENSE.md:195]: Codacy Issue found: [[list-item-content-indent] Don’t use mixed indentation for children, add 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md (hunk 1 above)
Comment (ricdezen) [inline LICENSE.md:130]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md [hunk 16]
```diff
@@ -0,0 +1,201 @@
...
+      and issue tracking systems that are managed by, or on behalf of, the
+      Licensor for the purpose of discussing and improving the Work, but
+      excluding communication that is conspicuously marked or otherwise
+      designated in writing by the copyright owner as "Not a Contribution."
+
+      "Contributor" shall mean Licensor and any individual or Legal Entity
+      on behalf of whom a Contribution has been received by Licensor and
+      subsequently incorporated within the Work.
+
+   2. Grant of Copyright License. Subject to the terms and conditions of
```
Comment (ricdezen) [inline LICENSE.md:66]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md (hunk 11 above)
Comment (ricdezen) [inline LICENSE.md:73]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md (hunk 16 above)
Comment (ricdezen) [inline LICENSE.md:66]: Codacy Issue found: [[list-item-bullet-indent] Incorrect indentation before bullet: remove 3 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

File: LICENSE.md (hunk 10 above)
Comment (ricdezen) [inline LICENSE.md:138]: Codacy Issue found: [[list-item-indent] Incorrect list-item indent: add 2 spaces](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4960568)

**Review by turcato-niccolo — APPROVED** [review]

**Comment by ricdezen** [comment]
Detto francamente: fanculo Codacy per questa volta. E' la licenza, non mi metto a modificarla.

**Review by bortoletti-giorgia — APPROVED** [review]

**Review by kummy98 — APPROVED** [review]

**Pull request merged**